libvips/foreign/jxlload.c

```diff
@@ -965,15 +960,37 @@ vips_foreign_load_jxl_header(VipsForeignLoad *load)
 				return -1;
 
 			if (JxlDecoderGetColorAsICCProfile(jxl->decoder,
-#ifndef HAVE_LIBJXL_0_9
-					&jxl->format,
-#endif
 					JXL_COLOR_PROFILE_TARGET_DATA,
 					jxl->icc_data, jxl->icc_size)) {
+				vips_foreign_load_jxl_error(jxl, "JxlDecoderGetColorAsICCProfile");
+				return -1;
+			}
+
+			/* Tinkering with output transforms. Experiment with this plus
+			 *
+			 *		https://sneyers.info/hdrtest/
+			 *
+			 * and see if we can get values out of SDR range
+			 */
+
+			JxlColorEncoding enc = {
+				.color_space = JXL_COLOR_SPACE_RGB,
+				.white_point = JXL_WHITE_POINT_D65,
+				.primaries = JXL_PRIMARIES_SRGB,
+				.transfer_function = JXL_TRANSFER_FUNCTION_SRGB,
+				.rendering_intent = JXL_RENDERING_INTENT_RELATIVE,
+			};
+			if (JxlDecoderSetPreferredColorProfile(jxl->decoder, &enc)) {
+				vips_foreign_load_jxl_error(jxl,
+					"JxlDecoderSetOutputColorProfile");
+				return -1;
+			}
+			if (JxlDecoderSetDesiredIntensityTarget(jxl->decoder, 10000)) {
```

Comment:
Should we revert this to its original?

Reply:
You're right, let's do the HDR stuff in a follow up PR.